src/simulation_objects/geometric/Polyhedron.cu

```diff
@@ -152,22 +153,33 @@ __host__ __device__ SpacePoint find_intersection_with_edge(SpacePoint a, SpacePo
 __host__ __device__ SpacePoint get_projected_vector_end(SpacePoint a, SpacePoint b, Face *current_face,
                                                         Polyhedron *polyhedron)
 {
+    SpacePoint vector_start = a;
+    SpacePoint vector_end = b;
```

Comment:
I don't like that when I call this function, the values I give to `a` and `b` are copied twice (firstly the values are copied to `a` and `b`, secondly they are copied to `vector_start` and `vector_end`). This is not a performance issue, because copying three `double`s is fast, moreover, I'm pretty sure this will be optimized during compilation, but I just don't like the current appearance of this piece of code